### PR TITLE
fix(releases): Support symbols in to_other argument

### PR DESCRIPTION
**Pull request by Zylphrex**
The to_other function currently allows for a string as the second argument. But
that string may contain various non alphanumeric characters that cause issues
with the aliases. This change adds support for a quoted string argument that
allows for these special characters and updates the aliases generation to handle
special characters.

#sync-getsentry

### CODE REVIEW

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -1417,7 +1417,7 @@ def get_json_meta_type(field_alias, snuba_type, function=None):
     return snuba_json
 
 
-FUNCTION_PATTERN = re.compile(r"^(?P<function>[^\(]+)\((?P<columns>[^\)]*)\)$")
+FUNCTION_PATTERN = re.compile(r"^(?P<function>[^\(]+)\((?P<columns>.*)\)$")
```
Comment (Zylphrex) [inline src/sentry/api/event_search.py:1420]: Arguments to `to_other` can include `)`, so only assert string ends with `)`, and anything in between is matched to columns.

File: src/sentry/api/event_search.py [hunk 2]
```diff
@@ -2344,12 +2355,14 @@ def get_function_alias(field):
...
-    return get_function_alias_with_columns(match.group("function"), columns)
+    function = match.group("function")
+    columns = parse_arguments(match.group("columns"))
+    return get_function_alias_with_columns(function, columns)
 
 
 def get_function_alias_with_columns(function_name, columns):
-    columns = "_".join(columns).replace(".", "_")
+    columns = re.sub("[^\w]+", "_", "_".join(columns))
+    columns = re.sub("_+", "_", columns).strip("_")
```
Comment (Zylphrex) [inline src/sentry/api/event_search.py:2365]: We want to substitute all special characters with `_` so the resulting alias is valid.

File: src/sentry/static/sentry/app/utils/discover/fields.tsx [hunk 3]
```diff
@@ -634,8 +634,9 @@ export function getAggregateAlias(field: string): string {
   }
   return field
     .replace(AGGREGATE_PATTERN, '$1_$2_$3')
-    .replace(/\./g, '_')
-    .replace(/\,/g, '_')
+    .replace(/[^\w]+/g, '_')
+    .replace(/_+/g, '_')
+    .replace(/^_+/g, '')
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/utils/discover/fields.tsx:639]: Updated to match the updated corresponding backend function.

File: src/sentry/static/sentry/app/views/releases/detail/overview/chart/utils.tsx [hunk 4]
```diff
@@ -36,17 +37,15 @@ export function getReleaseEventView(
   const {projects, environments, datetime} = selection;
   const {start, end, period} = datetime;
   const releaseFilter = currentOnly ? `release:${version}` : '';
-
-  const toOther = `to_other(release,${version},others,current)`;
-  // this orderby ensures that the order is [others, current]
-  const toOtherAlias = `to_other_release_${version}_others_current`;
+  const toOther = `to_other(release,"${escapeDoubleQuotes(version)}",others,current)`;
```
Comment (Zylphrex) [inline src/sentry/static/sentry/app/views/releases/detail/overview/chart/utils.tsx:40]: Make sure to quote and escape the version string so we work with special characters.

**Review by dashed — APPROVED** [review]

**Pull request merged**